app/assets/src/components/views/TaxonTreeVis.jsx

```diff
@@ -177,6 +177,7 @@ class TaxonTreeVis extends React.Component {
       let nodeId = taxon.tax_id;
       nodes.push({
         id: nodeId,
+        taxId: nodeId,
```

Comment:
why is taxId being set to nodeId here?

Reply:
**Short answer**: For these nodes (tax level 1 and 2, species and genus) node id is always equal to taxon id. I could just set id, and consider taxid to be equal to id when not set, but I believe it is cleaner this way.

**Long answer**:
Tax id and node can be different, but not always as in the case of some ancestors. They are different when the tax id in the data is a generic negative number. In that case the node id is composed of the generic tax id concatenated with a prefix with all the previous ancestors until the last known/unique one.

Tax id is going to be used to link other application places. 
Node id is going to be used for internal tree processing, and make sure that some nodes that have the same generic negative tax id, can be differentiated if they have different ancestry.